pkg/overlay/client.go

```diff
@@ -70,6 +70,28 @@ func (o *Overlay) Choose(ctx context.Context, amount int, space int64) ([]*pb.No
 	return resp.GetNodes(), nil
```

Comment:
I think we can replace the implementation of the `Choose` method with just:
```go
return o.ChooseFiltered(ctx, amount, space, nil)
```
This way we would avoid code duplication.